src/plugins/intel_cpu/src/nodes/mkldnn_matmul_node.cpp

```diff
@@ -425,6 +773,11 @@ void MKLDNNMatMulNode::prepareParams() {
         src1TransposedDesc = inDataDesc[1];
     }
 
+    matmul_kernel.reset();
+    prepareCustomKernel(src0TransposedDesc, src1TransposedDesc, *attr);
```

Comment:
Given that custom JIT kernel is not shape agnostic I don't see any value in enabling it for dynamic case. Lets move conditions and compilation part in CreatePrimitive().